tfhe/build.rs

```diff
@@ -73,7 +73,71 @@ fn gen_c_api() {
         .write_to_file(output_file);
 }
 
+fn panic_if_non_optimized_build() {
+    // This has a lot of \n and \ to control formatting precisely
+    const ERROR_MESSAGE: &str = "\
+    It seems that tfhe-rs is being compiled with insufficient optimization level.\n\
+    FHE is already slow on its own, and so compiler optimizations have to be turned on.\n\
+    \n\
+    By default rust/cargo does not enable optimizations, make sure you build/run with the \n\
+    release flag:\n    \
+    - `cargo run --release`\n    \
+    - `cargo build --release`\n\
+    (Note that doing `cargo build --release && cargo run`, will build in release mode\n\
+     but then rebuild in debug mode and run in debug mode)\n
+    \n\
+    * If you wish to be able to use a debugger, consider adding the following lines \n\
+    to the projects's Cargo.toml and compile in release\n\
+    \t```\n\
+    \t[profile.release]\n\
+    \tdebug = true\n\
+    \t```\n\
+    \t\n\
+    * It is also possible to override the opt-level only for tfhe-rs in the Cargo.toml\n\
```

Comment:
This could be the first suggestion as it is the less intrusive for a user (the long compilation time is limited to tfhe dependency)

In the same time, for some users unfamiliar to Rust, suggesting release profile is useful.

Maybe we could put this suggestion ("opt-level only for tfhe-rs") first and explain in the release profile suggestion that it's better to have all the code optimized to do performance tests

Reply:
I also think opt-level should be first because its needed for cargo check/clippy to work without having to
set --release all the time